### PR TITLE
Update Nintendo - Nintendo DS (No-Intro).json

### PR DESCRIPTION
**Pull request by retroNUC**
Adding onto this until you get the chance to review/merge:

- Add missing clone for "Dragon Quest VI"
- Add missing clone for "Puzzler World 2"
- Fix incorrect Japan region clone for "Cookie Shop - Create Your Dream Shop"
- Add missing clone for "Tamagotchi Connection - Corner Shop 2"
- Add new clone group for "I'm a Celebrity - Get Me Out of Here!"

### CODE REVIEW

File: clonelists/Nintendo - Nintendo DS (No-Intro).json [hunk 1]
```diff
@@ -1541,6 +1541,7 @@
 			"titles": [
 				{"searchTerm": "Dragon Quest VI - Maboroshi no Daichi"},
 				{"searchTerm": "Dragon Quest VI - Realms of Revelation"}
```
Comment (unexpectedpanda) [inline clonelists/Nintendo - Nintendo DS (No-Intro).json:1543]: This isn't valid JSON -- you need a comma at the end of the line here.

**Review by unexpectedpanda — COMMENTED** [review]
Don't forget to edit the `lastUpdated` value in the header too.

I'll check in on the `hash.json` value after the changes have been made.

File: clonelists/hash.json [hunk 2]
```diff
@@ -53,7 +53,7 @@
 	"Nintendo - Nintendo 3DS (No-Intro).json": "486d484cd3e464352941ea007035bde455e82c87269302099c9c12ee00c201ea",
 	"Nintendo - Nintendo 64 (No-Intro).json": "fe697b1a52506520d2cecaab2bd9ef1745cc90e8462024697988f33e3bc5c2a3",
 	"Nintendo - Nintendo 64DD (No-Intro).json": "5318a02a96be9e192a56732178c7212e4fff308b02fd4efa4750f40f93d52401",
-	"Nintendo - Nintendo DS (No-Intro).json": "1a62b75bf772e1666a9011c22806c3108a5437b420ea35420754ee1c8bee7856",
+	"Nintendo - Nintendo DS (No-Intro).json": "1a5ed51cdac75fc484bab2f3a9b31465e2753365afc42a0de2097ae54daee813",
```
Comment (unexpectedpanda) [inline clonelists/hash.json:56]: I'm getting a different hash here (`0838d86aee8f6b64abd2f102119b0eba0b3b59d27b8d8e77126df29e6057be01`). The only thing I can think of that might be messing things up are line endings -- they're meant to be purely LF (although I can't tell from this PR alone what line endings are in use).

It looks like I can force LF line endings on the repo itself, but that's not going to help calculating the hash in this file. For future submissions I'll update `hash.json` separately on my side until I can put tools and processes in people's hands to ensure consistency.

**Comment by retroNUC** [comment]
Yeah, I tried a few different combinations, but couldn't get the same hash as you. No worries.

Never used it before myself, but perhaps GitHub Actions could be used to generate `hash.json` consistently whenever file changes happen on the repo?

https://docs.github.com/en/actions/learn-github-actions/expressions#hashfiles

**Comment by unexpectedpanda** [comment]
Actions are the ultimate plan, one day :)

**Comment by andrebrait** [comment]
@unexpectedpanda do they have to be on the same commit? :)

**Comment by unexpectedpanda** [comment]
They can be different commits, but it can lead to a non-optimal outcome. Retool knows to download updates when the hash of the local file doesn't match the entry in `hash.json`. If the new file doesn't match that hash, the next time the user runs an update it'll download the same file again, even though an update isn't actually needed.

Realistically this is how it worked in the past. Contributors have only updated the clone list, I've merged, pulled the changes, converted, hashed, then committed the normalized content.

**Pull request merged**